service/gcs/bridge/bridge.go

```diff
@@ -189,22 +204,30 @@ type Bridge struct {
 
 	// testing hook to close the bridge ListenAndServe() method.
 	quitChan chan bool
+
+	protVer prot.ProtocolVersion
 }
 
 // AssignHandlers creates and assigns the appropriate bridge
 // events to be listen for and intercepted on `mux` before forwarding
 // to `gcs` for handling.
 func (b *Bridge) AssignHandlers(mux *Mux, gcs core.Core) {
 	b.coreint = gcs
-	mux.HandleFunc(prot.ComputeSystemCreateV1, b.createContainer)
-	mux.HandleFunc(prot.ComputeSystemExecuteProcessV1, b.execProcess)
-	mux.HandleFunc(prot.ComputeSystemShutdownForcedV1, b.killContainer)
-	mux.HandleFunc(prot.ComputeSystemShutdownGracefulV1, b.shutdownContainer)
-	mux.HandleFunc(prot.ComputeSystemSignalProcessV1, b.signalProcess)
-	mux.HandleFunc(prot.ComputeSystemGetPropertiesV1, b.listProcesses)
-	mux.HandleFunc(prot.ComputeSystemWaitForProcessV1, b.waitOnProcess)
-	mux.HandleFunc(prot.ComputeSystemResizeConsoleV1, b.resizeConsole)
-	mux.HandleFunc(prot.ComputeSystemModifySettingsV1, b.modifySettings)
+
+	// These are PvInvalid because they will be called previous to any protocol
+	// negotiation so they respond only when the protocols are not known.
+	mux.HandleFunc(prot.ComputeSystemNegotiateProtocolV1, prot.PvInvalid, b.negotiateProtocol)
+	mux.HandleFunc(prot.ComputeSystemCreateV1, prot.PvInvalid, b.createContainer)
```

Comment:
Currently, this means `createContainer` can't be called by a V4 HCS, because a protocol negotiation will set the version to a non-invalid value.

Reply:
Correct. After looking at it we will need a new impl for `createContainer` in v4 anyways so as a separate PR I will implement the new function.

Reply:
OK, sounds good!